Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -63,12 +63,7 @@ static bool validateCreateBuffer(const Device& device, const WGPUBufferDescripto
 
     // FIXME: "descriptor.usage is a subset of this.[[allowed buffer usages]]."
 
-    if ((descriptor.usage & WGPUBufferUsage_MapRead)
-        && (descriptor.usage & ~WGPUBufferUsage_CopyDst & ~WGPUBufferUsage_MapRead))
```

Comment:
Why the removal of `WGPUBufferUsage_CopyDst` & `WGPUBufferUsage_MapRead` constraints?

Reply:
The previous validation didn't allow for MapWrite | Vertex or other valid
combinations. I relaxed restrictions to only prevent buffers mapped as both
read and write which is invalid according to the specification.

I couldn't find where in the spec it says Read is only valid with CopyDst, does that exist? And if so, how would the buffers get bound as vertex / index / uniform / etc buffers?

Reply:
https://gpuweb.github.io/gpuweb/#buffer-usage

> If descriptor.[usage](https://gpuweb.github.io/gpuweb/#dom-gpubufferdescriptor-usage) contains [MAP_READ](https://gpuweb.github.io/gpuweb/#dom-gpubufferusage-map_read): descriptor.[usage](https://gpuweb.github.io/gpuweb/#dom-gpubufferdescriptor-usage) contains no other flags except [COPY_DST](https://gpuweb.github.io/gpuweb/#dom-gpubufferusage-copy_dst).
> 
> If descriptor.[usage](https://gpuweb.github.io/gpuweb/#dom-gpubufferdescriptor-usage) contains [MAP_WRITE](https://gpuweb.github.io/gpuweb/#dom-gpubufferusage-map_write): descriptor.[usage](https://gpuweb.github.io/gpuweb/#dom-gpubufferdescriptor-usage) contains no other flags except [COPY_SRC](https://gpuweb.github.io/gpuweb/#dom-gpubufferusage-copy_src).

The idea is that authors have to copy data in/out of all mappable buffers. I've tried many times to relax this restriction.

https://github.com/gpuweb/gpuweb/pull/2926#issuecomment-1261303560 

https://github.com/gpuweb/gpuweb/issues/2388